services/core-web/src/components/common/DocumentActions.tsx

```diff
@@ -0,0 +1,88 @@
+import React, { FC } from "react";
+import { bindActionCreators } from "redux";
+import { connect } from "react-redux";
+import { Dropdown, Menu, Button } from "antd";
+import { CARAT } from "@/constants/assets";
+import { DownloadOutlined, FileOutlined } from "@ant-design/icons";
+import { openDocument, isDocumentOpenable } from "@/components/syncfusion/DocumentViewer";
+import { downloadFileFromDocumentManager } from "@common/utils/actionlessNetworkCalls";
+
+interface DocumentActionsProps {
+  document: { documentName: string; documentMangerGuid: string };
+  openDocument: (arg1: string, arg2: string) => any;
+}
+
+export const DocumentActions: FC<DocumentActionsProps> = (props) => {
+  const downloadOnClick = (document) =>
+    document.documentMangerGuid
+      ? downloadFileFromDocumentManager({
+          document_manager_guid: document.documentMangerGuid,
+          document_name: document.documentName,
+        })
+      : null;
+
+  const canOpenInDocumentViewer = (document) =>
+    document.documentMangerGuid && isDocumentOpenable(document.documentName);
+
+  const openInDocumentViewerOnClick = (document) =>
+    document.documentMangerGuid
+      ? props.openDocument(document.documentMangerGuid, document.documentName)
+      : null;
+
+  const menu = (
+    <Menu>
+      {canOpenInDocumentViewer(props.document) && (
+        <Menu.Item key="0" icon={<FileOutlined />}>
+          <button
+            type="button"
+            className="full add-permit-dropdown-button"
+            onClick={() => {
+              openInDocumentViewerOnClick(props.document);
+            }}
+          >
+            <div>Open in Document Viewer</div>
+          </button>
+        </Menu.Item>
+      )}
+      <Menu.Item key="1" icon={<DownloadOutlined />}>
+        <button
+          type="button"
+          className="full add-permit-dropdown-button"
+          onClick={() => {
+            downloadOnClick(props.document);
+          }}
+        >
+          <div>Download File</div>
+        </button>
+      </Menu.Item>
+    </Menu>
+  );
+
+  return (
+    <div>
+      {/* @ts-ignore */}
+      <Dropdown overlay={menu} placement="bottomLeft">
```

Comment:
what are we ts-ignoring here?

Reply:
I just noticed that I'm getting (from a different component) console errors about this- `Warning: [antd: Dropdown] overlay is deprecated. Please use menu instead.` <-- can we use menu?

Reply:
Honestly I used ts-ignore here mainly because I noticed other files in the project would have the ts-ignore above the dropdown component. I thought it might of been because overlay was deprecated, but I found even if I were to remove the ts-ignore it still works with overlay? might be missing some context but I am not exactly too sure why ts-ignore was used.

As you suggested this has been changed to use menu instead of overlay now.